File: app/src/main/java/com/otraempresa/android/example/RequestSubscriptionTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;

import com.kushkipagos.android.Card;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;

class RequestSubscriptionTokenAsyncTask extends AbstractRequestTokenAsyncTask {

    RequestSubscriptionTokenAsyncTask(Context context, String currency) {
        super(context, currency);
    }

    @Override
    protected Transaction requestToken(Card card) throws KushkiException {
        return kushki.requestSubscriptionToken(card);
    }
}

File: app/src/main/java/com/kushkipagos/kushkidemo/services/RequestSubscriptionTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.services

import android.content.Context
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushkidemo.contracts.AbstractRequestTokenAsyncTask
import com.kushkipagos.models.Card
import com.kushkipagos.models.Transaction

class RequestSubscriptionTokenAsyncTask: AbstractRequestTokenAsyncTask {
    constructor(context: Context): super(context)

    @Throws(KushkiException::class)
    override fun requestToken(card: Card): Transaction {
        return _kushki.requestSubscriptionToken(card)
    }
}